src/esbmc/esbmc_parseoptions.h

```diff
@@ -76,6 +77,8 @@ class esbmc_parseoptionst : public parseoptions_baset, public language_uit
   FILE *out = stdout;
   FILE *err = stderr;
 
+  std::vector<goto_functions_algorithm> goto_preprocess_algorithms;
```

Comment:
This should contain pointers to accomodate objects of derived types, std::unique_ptr should fit here.

Reply:
Done